### PR TITLE
[AutoMM] Remove the config arg from fit API

### PR DESCRIPTION
**Pull request by zhiqiangdon**
*Issue #, if available:*

*Description of changes:*
Remove `config` from `fit` API to align with tabular and timeseries predictors. Users need to use `hyperparamters` instead.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by suzhoum — APPROVED** [review]
LGTM!

File: multimodal/tests/unittests/predictor/test_predictor.py [hunk 1]
```diff
@@ -440,149 +440,6 @@ def test_customizing_model_names(
             assert hasattr(predictor._learner._config.model, per_name)
 
 
-def test_model_configs():
```
Comment (taoyang1122) [inline multimodal/tests/unittests/predictor/test_predictor.py:443]: Do we need to add these to something like test_hyperparameters() ?

Comment (zhiqiangdon) [reply multimodal/tests/unittests/predictor/test_predictor.py:443]: This is to test passing a model config through `config`. The hyperparamters are already tested in other tests.

**Review by tonyhoo — APPROVED** [review]
LGTM, please make sure all tests passing before merge

**Pull request merged**